0.2/optimized properties

seanbarzilay opened the pull request:
#31

rmagen commented:
Bulk querying comes up in most of our Step implementations. We also need a simple way to configure the different bulk sizes and behaviors (e.g. Incremental bulk size). Maybe create a helper class for all of this?
A different Or maybe have a UniBulkStep that

seanbarzilay commented:
Fixed Properties strategy 
Deleted UniGraphVertexPropertiesSideEffectStep since it is no longer needed
Implemented UniBulkStep and converted steps to use it

rmagen commented on unipop-core/src/org/unipop/process/vertex/UniGraphVertexStep.java:
```diff
@@ -76,38 +87,47 @@ public UniGraphVertexStep(VertexStep<E> vertexStep, ControllerManager controller
+
+    private Iterator<Traverser.Admin<E>> getTraversersWithProperties(Iterator<Traverser.Admin<E>> traversers) {
+        List<Traverser.Admin<E>> copyTraversers = ConversionUtils.asStream(traversers).collect(Collectors.toList());
+        List<DeferredVertex> deferredVertices = copyTraversers.stream().map(traverser -> ((DeferredVertex) traverser.get())).collect(Collectors.toList());
```
Why is this still here?

seanbarzilay replied:
what do you mean?
